proto/minder/v1/minder.proto

```diff
@@ -2683,3 +2683,41 @@ message EvaluationHistoryAlert {
     // the structure and contents are alert specific, and are subject to change.
     string details = 2;
 }
+
+message EntityInstance {
+    // id is the unique identifier of the entity.
+    string id = 1;
+
+    // context is the context in which the entity is evaluated.
+    ContextV2 context = 2;
+
+    // name is the name of the entity.
+    string name = 3;
+
+    // type is the type of the entity.
+    // DISCUSSION: If we're aiming for a BYO entity type, we should probably
+    // have this be a string, and have the user provide the type.
+    Entity type = 4;
+
+    // properties is a map of properties of the entity.
+    google.protobuf.Struct properties = 5;
+}
+
+// UpstreamEntityRef providers enough information for the
+// provider to identify the entity in the upstream system.
+message UpstreamEntityRef {
```

Comment:
Do you think we should add a provider ID? I wonder if a common case for both `EntityInstance` and `UpstreamEntityRef` to instantiate its provider and then manipulate the data.

We could also add an internal struct that wraps these along with a provider ID and a project ID and maybe that's cleaner to not have those in the public API.

Reply:
@jhrozek we include provider information in `ContextV2`, I'm unsure if we need a provider ID in this case.

Reply:
sounds good!